map/hashtable.py

```diff
@@ -26,15 +27,14 @@ def put(self, key, value):
         initial_hash = hash_ = self.hash(key)
 
         while True:
-            if self._keys[hash_] is self._empty or self._keys[hash_] is self._deleted:
+            if not self._keys[hash_]:
```

Comment:
I found the issue with `m.get(0) == None`. Here this line should be reverted to the original version or `if self._keys[hash_] is None:`